docs/new-architecture-library-intro.md

```diff
@@ -0,0 +1,501 @@
+
+android/app/build.gradle:apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
+android/settings.gradle:apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
+...
+```
+
+### iOS
+
+On iOS, this generally requires your library to provide a Podspec (see [`react-native-webview`](https://github.com/react-native-community/react-native-webview/blob/master/react-native-webview.podspec) for an example).
+
+> To determine if your library is set up for autolinking, check the CocoaPods output after running `pod install` on an iOS project. If you see "auto linking library name", you are all set to go.
+
+## Preparing your JavaScript codebase for the new React Native Renderer (Fabric)
+
+The new renderer also known as Fabric doesn’t use the UIManager so direct calls to UIManager will need to be migrated. Historically, calls to UIManager had some pretty complicated patterns. Fortunately, we’ve created new APIs that are much cleaner. These new APIs are forwards compatible with Fabric so you can migrate your code today and they will work properly when you turn on Fabric!
+
+Fabric will be providing new type safe JS APIs that are much more ergonomic than some of the patterns we've seen in product code today. These APIs require references to the underlying component, no longer using the result of `findNodeHandle`. `findNodeHandle` is used to search the tree for a native component given a class instance. This was breaking the React abstraction model. `findNodeHandle` won’t be compatible with React 18 once we are ready to roll that out. Deprecation of `findNodeHandle` in React Native is similar to the [deprecation of `findDOMNode` in React DOM](https://reactjs.org/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage).
+
+While we know that all deprecations are a hassle, this guide is intended to help people update components as smoothly as possible. Here are the steps you need to take to get your JS codebase ready for Fabric:
+
+1. Migrating findNodeHandle / getting a HostComponent
+2. Migrating `.measure*()`
+3. Migrating off `setNativeProps`
+4. Move the call to `requireNativeComponent` to a separate file
+5. Migrating off `dispatchViewManagerCommand`
+6. Using `codegenNativeComponent`
+
+### Migrating `findNodeHandle` / getting a `HostComponent`
+
+<!-- alex ignore host -->
+
+Much of the migration work requires a HostComponent ref to access certain APIs that are only attached to host components (like View, Text, or ScrollView). HostComponents are the return value of calls to `requireNativeComponent`. `findNodeHandle` tunnels through multiple levels of component hierarchy to find the nearest native component.
+
+As a concrete example, this code uses `findNodeHandle` to tunnel from `MyComponent` through to the `View` rendered by `MyJSComponent`.
```

Comment:
*(Copy pasting my older comments because they're now on outdated files)*

`MyComponent` and `MyJSComponent` seem like unfortunate names in this example. Is the outer component not a "JS" component? It's hard to remember which one is which. I would suggest renaming to something like `Parent` and `Child` or `Panel` and `Button`, etc.

Reply:
Yup this was still on my table 👍 I'll close your [other comment](https://github.com/facebook/react-native-website/pull/2879#discussion_r768930761) in favour of this one as it's more up-to-date.

Reply:
Reworded to be `ParentComponent` vs `ChildComponent` by @sammy-SC here https://github.com/facebook/react-native-website/pull/2935